### DISCUSSION TITLE
Function prediction

### DISCUSSION THREAD

**Discussion by galkesten** [evolutionaryscale/esm#135] [Q&A]
What is the expected way to perform function prediction?

I read the `FunctionTokenizer` class and noticed that if we provide even one `FunctionAnnotation`, all other residues without a specified function are labeled with the `<none>` token. I also observed that the `<mask>` token is equal to the `<pad>` token, and when there are no function annotations, all residues are tokenized as `<pad>`, which is also treated as `<mask>`.

What does this mean? Are we supposed to make function predictions without providing known keyword annotations?

**Comment by JWGuo330** [comment]
I also have the same question
